source/mainnet/smart-contracts/onboarding-guide-ethereum-developers/faq.rst

```diff
@@ -0,0 +1,760 @@
+
+    .. code-block:: console
+
+        $concordium-client contract update <ContractIndex> --entrypoint <ContractEntryPoint> --parameter-binary myInputParameters.bin --sender <Account> --energy 12345678
+
+    You can create such a `myInputParameters.bin` file by adding the below lines to your test cases replacing the `ExampleParams` struct with your input parameter struct for that function.
+
+    .. code-block:: console
+
+        let parameter = ExampleParams {
+            example_key1: value1,
+            example_key1: value2,
+        }
+        let parameter_bytes = to_bytes(&parameter);
+        std::fs::write("myInputParameters.bin", &parameter_bytes).expect("Failed to write parameter file");
+
+    When running the tests with the below command the `myInputParameters.bin` file is created in the current folder.
+
+    .. code-block:: console
+
+        $ cargo test
+
+Events:
+=======
+
+.. dropdown::  Where can I find a logged event on testnet/mainnet?
+
+    You can look up a tx hash on the `dashboard <https://dashboard.testnet.concordium.com/lookup/13ded9aaf6085e970b2cf3874431de5805ffa35a553c93707d1863a8888e8aa4>`_.
+    It will provide you with the full execution chain of the smart contracts that
+    were invoked and updated during this tx.
+    You can click on an updated contract instance row to see additional information.
+    For example, navigate to the last page (third page) of the displayed execution chain of `this tx <https://dashboard.testnet.concordium.com/lookup/13ded9aaf6085e970b2cf3874431de5805ffa35a553c93707d1863a8888e8aa4>`_
+    and click on the top row `Updated contract instance at address: <783,0>`. You will see additional information
+    about the smart contract address, name, the function entry point that was invoked,
+    the CCD amount that was sent to the function, and events that were logged by this smart contract function.
+
+    The below picture shows that one event was logged by the contract `<783,0>` and no event was logged by the contract `<782,0>`.
+
+    .. image:: ./images/onboarding_ethereum_developers_1.png
+        :width: 100 %
+
+    If several events are logged by one function entry point, the different events can be distinguished by their array index.
+    The below picture shows that four events were logged by an entry point in
+    `this execution chain <https://dashboard.testnet.concordium.com/lookup/7fcad417384d8e36fd2264d16d0ce1385860cdad711d17f7d6c12137c9cbab2e>`_.
+
+    .. image:: ./images/onboarding_ethereum_developers_2.png
+        :width: 100 %
+
+.. dropdown::  How can I compare/decode the event data that I see on the dashboard?
+
+    A smart contract developer is free to use their own way of encoding and
+    interpreting event log data when writing their own smart contracts.
+
+    The official Concordium smart contracts use the following standard:
+    Each event has a number tag then two 0s and the rest of the event data.
```

Comment:
Where do the two 0's come from?

Reply:
I would guess the two zeros are from all seeing the wCCD events, since the token ID is `00` in hex.
So this is not true for a token with a different token ID.